### PR TITLE
🆕 Use Annotations as a Filter for Patch Extraction

### PR DESCRIPTION
**Pull request by measty**
This PR is something i've been thinking about from a conversation with Daan when he was here from Radboud doing a seminar a while back. His comment was that one thing some of his colleges felt was missing in TIAtoolbox was ability to get patches filtered by class/label.

Its not that hard to set up tiatoolbox patch extractor to allow this, which is what i've added in this PR.

it allows you to provide a path to an annotation store as the mask. By default the mask will then be built out of all the annotations, but you can also provide any filter that works in the annotation store, and have only those annotations make the mask. So for example, if you have some pathologist annotations with some regions, and you only want to extract patches which have some overlap with regions labelled 'tumor' or whatever, you would just have to provide the appropriate string ("props['label'] == 'tumor'" for example).

### CODE REVIEW

File: tiatoolbox/tools/patchextraction.py [hunk 1]
```diff
@@ -48,6 +49,7 @@ class ExtractorParams(TypedDict, total=False):
     input_mask: str | Path | np.ndarray | wsireader.VirtualWSIReader
```
Comment (shaneahmed) [inline tiatoolbox/tools/patchextraction.py:49]: Update docstring for input_mask as it now accepts "AnnotationStore"

**Review by shaneahmed — APPROVED** [review]
Thanks @measty

**Comment by shaneahmed** [comment]
> Thanks @measty

@mostafajahanifar Please can you review this PR so we can merge this.

**Comment by shaneahmed** [comment]
@mostafajahanifar I have updated the file.

**Review by mostafajahanifar — APPROVED** [review]
Well done, Mark! Great PR 💯 

I only made some minor comments on the Notebook, feel free to make them or to ignore them.

**Pull request merged**